### DISCUSSION TITLE
Fitting to MD Data Harder than Suggest Mode?

### DISCUSSION THREAD

**Discussion by ejmeitz** [ttadano/alamode#236] [Q&A]
Should fitting to MD data give worse fits than fitting to the suggested displacements? I've been trying to replicate a fit to SW silicon and cannot get error better than 85%. The displacement (suggest) approach works fine, but when I generate the DFSET myself things do not work so well. I've spent a rather long time trying to find a bug in my approach and cannot. Is there any things I should be aware of when using MD data instead of displacements? Does the fitting scheme have issues when multiple atoms are moving? I've used other IFC packages with similar data and it has worked fine so I expect a bug in my implementation but I really do not see what could be wrong. I've ensured that that order of atoms given to &position are the same order as the DFSET and that units are correct among other things.

**Comment by ttadano** [comment]
If you fit the force constants to the displacement-force datasets extracted from a MD run, the fitting error usually becomes larger as you mentioned. However, the error of 85% looks too large. Perhaps, something is wrong with the inputs.

When you convert the atomic positions in the supercell into displacements, did you treat the periodic boundary condition properly? Did you use `extract.py` for generating the DFSET?

**Reply by ejmeitz** [reply]
I did not use extract.py, I just manually made the DFSET since I was trying to mix samples from multiple MD trajectories (in testing I just used 1 MD simulation though). I have a lammps script which computes `displace/atom` and then dumps that along with fx fy fz to a dump file (sorted by id). I also tried dumping xu, yu and zu and manually calculating displacements. The boundary conditions should be accounted for correctly.

One weird thing I noticed is that the dispersion and DOS look alright even though the fit is very bad. Values from the anharmonic IFCs (like Gruneissen parameter and thermal conductivity) are very far from correct though.

How big of a difference would you expect for the error? I was testing at 100K for Silicon which should be extremely harmonic and fit well by just second-order IFCs.

**Reply by ttadano** [reply]
I've checked my old results where I used AIMD trajectories to fit the force constants of Si.

Even at 1000 K, the fitting error was only 29% when I only considered harmonic IFCs, and the error reduced to 10% when I included third-order IFCs within the third-nearest neighbors.

So, the large error of 85% even at 100 K indicates some issues with the dataset.

**Reply by ejmeitz** [reply]
Thanks Ill look more for a bug in my implementation then.

Also from this: https://github.com/ttadano/alamode/discussions/233#discussioncomment-11219272
should FC2XML keyword work with XML files generated by ALM? Mine do not seem to work unless they're from ALAMODE.
